internal/reader/onchain_prices_reader.go

```diff
@@ -18,55 +24,60 @@ type TokenPrices interface {
 	GetTokenPricesUSD(ctx context.Context, tokens []ocr2types.Account) ([]*big.Int, error)
 }
 
-type TokenPriceConfig struct {
-	// This is mainly used for inputTokens on testnet to give them a price
-	StaticPrices map[ocr2types.Account]big.Int `json:"staticPrices"`
-}
-
 type OnchainTokenPricesReader struct {
-	TokenPriceConfig TokenPriceConfig
 	// Reader for the chain that will have the token prices on-chain
-	ContractReader commontypes.ContractReader
+	ContractReader commontyps.ContractReader
+	PriceSources   map[types.Account]pluginconfig.ArbitrumPriceSource
 }
 
 func NewOnchainTokenPricesReader(
-	tokenPriceConfig TokenPriceConfig, contractReader commontypes.ContractReader,
+	contractReader commontyps.ContractReader,
+	priceSources map[types.Account]pluginconfig.ArbitrumPriceSource,
 ) *OnchainTokenPricesReader {
 	return &OnchainTokenPricesReader{
-		TokenPriceConfig: tokenPriceConfig,
-		ContractReader:   contractReader,
+		ContractReader: contractReader,
+		PriceSources:   priceSources,
 	}
 }
 
+type LatestRoundData struct {
+	RoundID         *big.Int
+	Answer          *big.Int
+	StartedAt       *big.Int
+	UpdatedAt       *big.Int
+	AnsweredInRound *big.Int
+}
+
 func (pr *OnchainTokenPricesReader) GetTokenPricesUSD(
 	ctx context.Context, tokens []ocr2types.Account,
 ) ([]*big.Int, error) {
-	const (
-		contractName = "PriceAggregator"
-		functionName = "getTokenPrice"
-	)
 	prices := make([]*big.Int, len(tokens))
 	eg := new(errgroup.Group)
 	for idx, token := range tokens {
 		idx := idx
 		token := token
 		eg.Go(func() error {
-			price := new(big.Int)
-			if staticPrice, exists := pr.TokenPriceConfig.StaticPrices[token]; exists {
-				price.Set(&staticPrice)
-			} else {
-				if err :=
-					pr.ContractReader.GetLatestValue(
-						ctx,
-						contractName,
-						functionName,
-						primitives.Finalized,
-						token,
-						price); err != nil {
-					return fmt.Errorf("failed to get token price for %s: %w", token, err)
-				}
+			//TODO: Once chainreader new changes https://github.com/smartcontractkit/chainlink-common/pull/603
+			// are merged we'll need to use the bound contract
+			//boundContract := commontypes.BoundContract{
+			//	Address: pr.PriceSources[token].AggregatorAddress,
+			//	Name: consts.ContractNamePriceAggregator,
+			//}
+
+			latestRoundData := LatestRoundData{}
+			if err :=
+				pr.ContractReader.GetLatestValue(
+					ctx,
+					consts.ContractNamePriceAggregator,
+					consts.MethodNameGetLatestRoundData,
+					primitives.Finalized,
+					nil,
+					latestRoundData,
+					//boundContract,
+				); err != nil {
+				return fmt.Errorf("failed to get token price for %s: %w", token, err)
 			}
-			prices[idx] = price
+			prices[idx] = latestRoundData.Answer
```

Comment:
There's a couple of things we need to check before we return this price; we need to make sure that the price is in USD with 18 decimals, per 1e18 of the smallest token denomination. See https://github.com/smartcontractkit/ccip/blob/89519db20808d4210c1f9025d1355fd0b0076585/contracts/src/v0.8/ccip/PriceRegistry.sol#L152-L157 for details.